### PR TITLE
Ens contracts submodule

### PR DESCRIPTION
**Pull request by Siegrift**
Name bazaar is dependent on ENS contracts (ensdomains/ens repository). This PR adds the the repo as a submodule
fixed at a specific version instead of copied contracts from ens. This should improve contract migrations in the
future, because one can just bump the version of submodule.
We have also reflected these changes in README and updated gitignore to avoid commiting generated files.

I've inherited from ens Registry contract and placed the custom dev method in our new subclass, which I then used in the project. The only changes in the code were replacements of :registrar with :name-bazaar-registrar and using this smart contract in the smart-contracts.cljs.

Matus agreed to ignore *.abi files. So I've excluded them as well.

Unfortunately, there is not a single clojure formatter which we could use as a tool. I ran cursive formatted on all source files in the project. Not sure if we want to merge it, but we can easily revert the last commit if necessary. But to me these changes make the code more readble.

P.S. I could probably avoid the formatting changes if I redo the find/replace in the clojure source files again. But I am not sure if that is worth the cost.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "resources/public/contracts/src/ens-repo"]
+	path = resources/public/contracts/src/ens-repo
+	url = https://github.com/ensdomains/ens.git
```
Comment (ladislavdubravsky) [inline .gitmodules:3]: I take it it's not possible to have a specific commit in this file?

File: README.md [hunk 2]
```diff
@@ -10,6 +10,12 @@ Smart-contracts can be found [here](https://github.com/district0x/name-bazaar/tr
 
 ## Starting a dev server with an Ethereum Testnet
 
+First, clone the project with submodules _([ens](https://github.com/ensdomains/ens))_
```
Comment (ladislavdubravsky) [inline README.md:13]: What if you have cloned the repository before and want to just load the submodules now, i.e. position I'm in when reviewing now?
I gathered I should use:
`git submodule update --init --recursive`
Please add this case to README as well.

File: .gitmodules (hunk 1 above)
Comment (Siegrift) [reply .gitmodules:3]: In this file no, this is just an information about which repo is the submodule. However, the submodule acts as a normal git repo inside the project - Meaning that it has notion of branches and commits internally and the branch of submodule is fixed in the project. If you bump the submodule commit, you WILL commit it as change (smthng like this https://github.com/Siegrift/react-diagrams-app/commit/124d411300f0157814765023f458fbecc36872fd#diff-e7e55e6a474cad651590585e3553473c930a9b73f8582b6ff99e95a0b698dd52)

File: README.md [hunk 3]
```diff
@@ -90,6 +99,20 @@ or
...
+`contracts/ens-repo` directory, you can use
+
+```bash
+git submodule update --remote
+git log --pretty=oneline --reverse --ancestry-path HEAD..master
+```
+
+to first update the the ens repository and then list the commits in ens `master` branch, which are
+not in our version of `ens-repo`. _(You can use this list of commits when updating to the latest
+version of ens contracts)_
```
Comment (ladislavdubravsky) [inline README.md:121]: Do you have to `git checkout something` in the ENS repository after this and then commit in the parent repository?
Regardless of yes/no answer I'd explicitly mention the workflow for checking out a particular commit / branch on ENS and commiting that in parent repository.

**Comment by ladislavdubravsky** [comment]
The tests fail for me after:
```
> git checkout ens-contracts-submodule
> git submodule update --init --recursive
Submodule path 'resources/public/contracts/src/ens-repo': checked out '6c0cf08eab71b41d54ac516138d4e1f2c32ff5c9'
> solc use 0.4.18
> solc --version
Version: 0.4.18+commit.9cf6e910.Linux.g++
> docker run --name=namebazaar-ganache -d -p 8549:8549 trufflesuite/ganache-cli:v6.12.1 -p 8549
> curl --location --request POST 'localhost:8549' --header 'Content-Type: application/json' --data-raw '{ "jsonrpc": "2.0", "method": "eth_blockNumber", "params": [],"id": 1 }'
{"id":1,"jsonrpc":"2.0","result":"0x0"}
> ./compile-solidity.sh
> lein doo node "server-tests"
...
FAIL in (contracts-setup) (cljs/test.cljs:476:9)
expected: (= (contract-address :ens) (registrar/ens))
  actual: (not (= "0x5c0a54e5d85d7422f47c0a4d877fe984aac1bf12" "0x"))
...
```

File: README.md (hunk 3 above)
Comment (Siegrift) [reply README.md:121]: Yes, you need. I will maybe add a section how to work with submodules as that requires a bit of knowledge.

**Comment by ladislavdubravsky** [comment]
Manual clean of build folders fixed this for me. I created an [issue](https://github.com/vacuumlabs/name-bazaar/issues/3) for that problem. Concerning this PR, LGTM.

**Review by ladislavdubravsky — APPROVED** [review]

**Pull request merged**